otx/cli/manager/config_manager.py

```diff
@@ -243,6 +241,96 @@ def auto_task_detection(self, data_roots: str) -> str:
+        """
+
+        def _check_is_only_images(dir):
+            """Check if a directory contains only images."""
+            import glob
+
+            pngs = glob.glob(f"{dir}/*.png")
+            jpgs = glob.glob(f"{dir}/*.jpg")
+            return len(pngs) > 0 or len(jpgs) > 0
+
+        def _check_semisl_requirements(train_dir, unlabeled_dir, thershold=0.07):
+            """Check if quantity of unlabeled images is sufficient for Semi-SL learning."""
+            all_unlabeled_images = len(os.listdir(unlabeled_dir))
+            if all_unlabeled_images <= 1:
+                # the folder is empty or only 1 image there
+                return False
+            all_train_images = len(os.listdir(train_dir))
+            # check if number of unlabeled images is more than relative thershold
+            if all_unlabeled_images >= thershold * all_train_images:
+                return True
+            return False
+
+        # if user explicitly passed train type via args
+        if self.args.train_type is not None:
+            self.train_type = self.args.train_type
+            return
+
+        path_to_train_data = Path(self.args.train_data_roots)
+        if not Path.is_dir(path_to_train_data):
+            raise ValueError(
+                "train-data-roots isn't a directory or it doesn't exist. "
+                "Please, check command line and directory path."
+            )
+
+        if _check_is_only_images(path_to_train_data):
+            # If train folder with images only was passed to args
+            # Then we start self-supervised training
+            print("[*] Selfsupervised training type detected")
+            self.train_type = "Selfsupervised"
+            return
+
+        # if user explicitly passed unlabeled images folder
+        if self.args.unlabeled_data_roots is not None:
+            unlabeled_path = Path(self.args.unlabeled_data_roots)
+            if Path.is_dir(unlabeled_path):
+                if not _check_semisl_requirements(path_to_train_data, unlabeled_path):
+                    print(
+                        "WARNING: There are none or too litle images to start Semi-SL training."
+                        "It should be more that relative threshold"
+                        "Start Supervised training instead."
+                    )
+                self.train_type = "Semisupervised"
+                return
+            else:
+                raise ValueError(
+                    "Unlabeled images isn't a directory or it doesn't exist. "
+                    "Please, check command line and directory path."
+                )
+
+        unlabeled_folder_name = [
+            item
+            for item in os.listdir(path_to_train_data)
+            if (item.startswith("unlabeled") and os.path.isdir(os.path.join(path_to_train_data, item)))
+        ]
+        if unlabeled_folder_name:
+            # if in directory we have several unlabeled data folders -> retrive first
```

Comment:
Are there any reason of only retrieving first? How about getting all images if there are more than one directory.

Reply:
It is a complicated scenario I think, because in that case we should copy all images in one place and then pass the path further. If we have large number of images - it will be time consuming, unfortunately. Because of that, I use only the first found folder.